### PR TITLE
Extract mentions from comments & bounties 

### PR DESCRIPTION
**Pull request by Devorein**
1. Extract mentions from page comment
2. Extract mentions from workspace bounties
3. Skip self mentions
4. Link to correct mention for bounty

### CODE REVIEW

File: charmClient.ts [hunk 1]
```diff
@@ -290,7 +290,7 @@ class CharmClient {
 
     return contributors.map((contributor: Contributor) => ({
       id: contributor.id,
-      username: contributor.username,
+      username: contributor.username as string,
```
Comment (mattcasey) [inline charmClient.ts:293]: this is redundant, maybe you started before Andre made this a required field

File: components/common/CharmEditor/InlineCharmEditor.tsx [hunk 2]
```diff
@@ -15,12 +15,15 @@ import { CSSProperties, ReactNode, useState } from 'react';
...
 import { PageContent } from 'models';
+import { useCurrentSpace } from 'hooks/useCurrentSpace';
+import { useUser } from 'hooks/useUser';
 import FloatingMenu, { floatingMenuPlugin } from './components/FloatingMenu';
 import EmojiSuggest, * as emoji from './components/emojiSuggest';
 import Mention, { mentionPlugins, mentionSpecs, MentionSuggest, mentionPluginKeyName } from './components/mention';
 import * as tabIndent from './components/tabIndent';
 import Placeholder from './components/Placeholder';
 import { checkForEmpty } from './utils';
+import { charmPlugin } from './components/charm/charm.plugins';
```
Comment (mattcasey) [inline components/common/CharmEditor/InlineCharmEditor.tsx:26]: what does this plugin do? can you come up with a better name or add a description?

File: lib/mentions/interfaces.ts [hunk 3]
```diff
@@ -1,14 +1,54 @@
...
+  bountyTitle: string
+  commentId: string | null
+  mentionId: string
+  createdAt: string
   createdBy: User | null
+  text: string
+  type: 'bounty'
+}
+
+export interface CommentMentionedTask {
```
Comment (mattcasey) [inline lib/mentions/interfaces.ts:37]: why is a comment its own type of task? I think conceptually, Bounty and Page mentions would just have an optional commentId...

File: components/common/Avatar.tsx [hunk 4]
```diff
@@ -29,7 +29,7 @@ const StyledAvatar = styled(Avatar)`
 export type InitialAvatarProps = {
   avatar: string | null | undefined;
   className?: string;
-  name?: string;
+  name?: string | null;
```
[the remark this answers is not in the document]
Comment (Devorein) [reply components/common/Avatar.tsx:32]: It was the same issue as above. Reverted back the changes

**Pull request merged**